### PR TITLE
Port SQL transactionality tests to logic_test

### PR DESCRIPTION
**Pull request by tamird**
This moves most of the tests added in 568de717ca9147345c483e5c374dfd16e447ec6b to the logic test suite.

`create_test.go` is not moved because it's still needed to validate transactionality with respect to `DescIDGenerator` which is not included in the SQL system tables.

### CODE REVIEW

File: sql/logic_test.go [hunk 1]
```diff
@@ -436,7 +437,25 @@ func (t *logicTest) execQuery(query logicQuery) {
...
+		}
+		fmt.Fprint(w, "but found:\n")
+		for i := 0; i < len(results); i += len(cols) {
+			for _, value := range results[i : i+len(cols)] {
+				fmt.Fprintf(w, "%q\t", value)
+			}
+			fmt.Fprint(w, "\n")
+		}
+		w.Flush()
+		t.Fatal(buf.String())
```
Comment (petermattis) [inline sql/logic_test.go]: It's nice that this is a separate commit, but really this should be a separate PR as it is unrelated to the conversion of the tests.

File: sql/testdata/explain_debug [hunk 2]
```diff
@@ -49,6 +53,38 @@ UPDATE abc SET c = NULL WHERE a = 1
...
-1  /abc/primary/2/'two'    NULL  true
-2  /abc/primary/3/'three'  NULL  true
+0  /abc/primary/1/'one'      NULL  true
+1  /abc/primary/2/'two'      NULL  NULL
+1  /abc/primary/2/'two'/c    2.2   true
+2  /abc/primary/3/'three'    NULL  NULL
+2  /abc/primary/3/'three'/c  3.3   true
+
+statement error duplicate key value .* violates unique constraint
+INSERT INTO abc VALUES (4, 'four', 3.3, NULL)
```
Comment (petermattis) [inline sql/testdata/explain_debug]: Shouldn't this test, and the ones below for `UPDATE` and `DELETE` belong in a separate file. Placing them in `testdata/explain_debug` feels convenient but inappropriate. Perhaps `testdata/unique_index`.

File: sql/testdata/explain_debug [hunk 3]
```diff
@@ -49,6 +53,38 @@ UPDATE abc SET c = NULL WHERE a = 1
...
+0  /abc/primary/1/'one'      NULL  true
+1  /abc/primary/2/'two'      NULL  NULL
+1  /abc/primary/2/'two'/c    2.2   true
+2  /abc/primary/3/'three'    NULL  NULL
+2  /abc/primary/3/'three'/c  3.3   true
+
+statement ok
+DELETE FROM abc
+
+EXPLAIN (DEBUG) SELECT * FROM abc
```
Comment (petermattis) [inline sql/testdata/explain_debug]: You've forgotten the `query` command. I don't think this is doing anything.

File: sql/testdata/explain_debug [hunk 4]
```diff
@@ -3,9 +3,10 @@ CREATE TABLE abc (
   a INT,
   b TEXT,
   c FLOAT,
+  d CHAR,
   PRIMARY KEY (a, b),
-  CONSTRAINT foo UNIQUE (b),
-  CONSTRAINT bar INDEX (a)
+  CONSTRAINT foo UNIQUE (c),
```
Comment (petermattis) [inline sql/testdata/explain_debug]: Was there a fix to indexing of float columns that I missed?

**Comment by petermattis** [comment]
LGTM

File: sql/testdata/explain_debug (hunk 4 above)
Comment (tamird) [inline sql/testdata/explain_debug]: indexing of floats was never the problem - writing NULL to any indexed column is the problem case, which i decided to side-step in this PR

File: sql/logic_test.go (hunk 1 above)
Comment (tamird) [inline sql/logic_test.go]: if you really feel that way, I'm happy to do it; it made my life much easier when I had to modify the explain_debug tests, and I did it as a prerequisite to the other changes. I don't agree that there's any value in putting in a separate PR, though, unless it's a controversial change that is going to receive separate discussion.

File: sql/testdata/explain_debug (hunk 3 above)
Comment (tamird) [inline sql/testdata/explain_debug]: Thanks, fixed.

File: sql/testdata/explain_debug (hunk 2 above)
Comment (tamird) [inline sql/testdata/explain_debug]: Sounds good. I'll put these in a separate file `testdata/atomic` since the intention was to test transactionality of single statements

Comment (tamird) [inline sql/testdata/explain_debug]: actually, i'll fold them into the existing test/update/delete tests; i don't think that's inappropriate

File: sql/logic_test.go (hunk 1 above)
Comment (petermattis) [inline sql/logic_test.go]: The value of separate PRs for unrelated changes is that it lowers the burden on the reviewer, both from a lines of code to review perspective and from a conceptual perspective.

There is zero value to separating this out now. Please don't waste any effort doing so.

File: sql/testdata/explain_debug (hunk 4 above)
Comment (petermattis) [inline sql/testdata/explain_debug]: File an issue?

Comment (tamird) [inline sql/testdata/explain_debug]: yep

File: sql/logic_test.go (hunk 1 above)
Comment (tamird) [inline sql/logic_test.go]: OK. My intention in separating it into its own commit was to lighten the reviewer's burden without incurring the overhead of an extra PR. Perhaps I should make that more clear.

**Comment by tamird** [comment]
Updated, PTAL

**Comment by petermattis** [comment]
LGTM
